crates/pipeline_manager/src/pipeline_manager.rs

```diff
@@ -603,96 +594,98 @@ struct ProgramCodeResponse {
     code: String,
 }
 
-/// Returns the latest SQL source code of the program along with its meta-data.
+/// Fetch programs, optionally filtered by name or ID.
 #[utoipa::path(
     responses(
-        (status = OK, description = "Program data and code retrieved successfully.", body = ProgramCodeResponse),
-        (status = BAD_REQUEST
-            , description = "Specified program id is not a valid uuid."
+        (status = OK, description = "Programs retrieved successfully.", body = [ProgramDescr]),
+        (status = NOT_FOUND
+            , description = "Specified program name does not exist in the database."
             , body = ErrorResponse
-            , example = json!(example_invalid_uuid_param())),
+            , example = json!(example_unknown_name())),
         (status = NOT_FOUND
```

Comment:
I thought two `NOT_FOUND` statuses are not supported. Don't you need to merge them?